src/test/java/com/rocketden/main/service/GameManagementServiceTests.java

```diff
@@ -154,13 +160,62 @@ public void getGameSuccess() {
 
         GameDto gameDto = gameService.getGameDtoFromRoomId(ROOM_ID);
 
-        assertEquals(RoomMapper.toDto(room), gameDto.getRoomDto());
-        assertNull(gameDto.getPlayerMap());
+        assertEquals(RoomMapper.toDto(room), gameDto.getRoom());
+        assertEquals(1, gameDto.getPlayers().size());
+        assertEquals(UserMapper.toDto(user), gameDto.getPlayers().get(0).getUser());
     }
 
     @Test
     public void getGameNotFound() {
         ApiException exception = assertThrows(ApiException.class, () -> gameService.getGameDtoFromRoomId(ROOM_ID));
         assertEquals(GameError.NOT_FOUND, exception.getError());
     }
+
+    @Test
+    public void submitSolutionSuccess() {
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        User user = new User();
+        user.setNickname(NICKNAME);
+        user.setUserId(USER_ID);
+        room.addUser(user);
+
+        Game game = gameService.createAddGameFromRoom(room);
+
+        SubmissionRequest request = new SubmissionRequest();
+        request.setLanguage(LANGUAGE);
+        request.setCode(CODE);
+        request.setInitiator(UserMapper.toDto(user));
+
+        gameService.submitSolution(ROOM_ID, request);
+
+        verify(submitService).submitSolution(game, request);
+    }
+
+    @Test
+    public void submitSolutionFailure() {
```

Comment:
Could you test the empty field error (`GameError.EMPTY_FIELD`) as well?

Reply:
Sounds good - I did have a small test for that at the bottom of this method but realized that might be confusing/messy so I moved it to a separate test